### PR TITLE
Added TSStepName Parameter to Show-TSErrorDialog

### LINKED ISSUE #1: ShowErrorDialog adds parameter in TP1902.2
In TP 2902.2 ShowErrorDialog added an 8th parameter to the com object call.

I don't have a readily available environment so getting around to this may not be super quick. Being that this is in TP there's always a chance this doesn't make it to production :wink:

I'd rather have some context to what the parameter is used for. If anyone is interested, inspecting the dll for the exported functions may give us a parameter name. That might be easier than passing a value to that parameter slot and seeing what happens 🤷‍♂️ 

Reported via [twitter](https://twitter.com/gwblok/status/1105213253712576514) by @AdamGrossTX

### PR DESCRIPTION
**Pull request by AdamGrossTX**
Added TSStepName Parameter to Show-TSErrorDialog for SCCM 1901 TP and newer support,

fixes #1

### CODE REVIEW

File: SCCM-TSEnvironment.psm1 [hunk 1]
```diff
@@ -440,19 +445,24 @@ function Show-TSErrorDialog()
...
-        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, 1)
+    [int]$Reboot = Switch($ForceReboot) {
+        $True {1}
+        $False {0}
     }
-    else
-    {
-        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, 0)
+
+    If([string]::IsNullOrEmpty($TSStepName)) {
```
Comment (sombrerosheep) [inline SCCM-TSEnvironment.psm1]: For the sake of consistency--can your `If`/`Else`/`True`/`False` match the case of the file and be `if`/`else`/`true`/`false`?  (yeah i know.. :smile:)

File: SCCM-TSEnvironment.psm1 [hunk 2]
```diff
@@ -440,19 +445,24 @@ function Show-TSErrorDialog()
...
+        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, $Reboot)
+    }
+    Else {
+        $Script:TaskSequenceProgressUi.ShowErrorDialog($OrganizationName, $Script:TaskSequenceEnvironment.Value("_SMSTSPackageName"), $CustomTitle, $ErrorMessage, $ErrorCode, $TimeoutInSeconds, $Reboot, $TSStepName)
```
Comment (sombrerosheep) [inline SCCM-TSEnvironment.psm1:465]: Would you be able to provide a visual of "where" this value shows up in the dialog? If I remember there were a few things that (while requiring a parameter) they didn't really "do anything".

File: SCCM-TSEnvironment.psm1 [hunk 3]
```diff
@@ -427,6 +428,10 @@ function Show-TSErrorDialog()
     Sets an Error and forces a reboot
     Show-TSErrorDialog -OrganizationName "My Organization" -CustomTitle "An Error occured during the things" -ErrorMessage "He's dead Jim!" -ErrorCode 123456 -TimeoutInSeconds 90 -ForceReboot $true
 
+    .EXAMPLE
+    Adds TSStepName which is required for SCCM 1901 TP and newer
+    Show-TSErrorDialog -OrganizationName "My Organization" -CustomTitle "An Error occured during the things" -ErrorMessage "That thing you tried...it didnt work" -ErrorCode 123456 -TimeoutInSeconds 90 -ForceReboot $false -TSStepName "My Step Name"
```
Comment (sombrerosheep) [inline SCCM-TSEnvironment.psm1:433]: Think is beneficial to recommend `_SMSTSCurrentActionName` as a good default parameter?

Comment (AdamGrossTX) [reply SCCM-TSEnvironment.psm1:433]: Yes. I think that makes sense.

File: SCCM-TSEnvironment.psm1 (hunk 1 above)
Comment (AdamGrossTX) [reply SCCM-TSEnvironment.psm1]: fixing in the next PR

**Review by sombrerosheep — APPROVED** [review]

**Pull request merged**